src/shogun/lib/type_case.h

```diff
@@ -0,0 +1,369 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#ifndef SHOGUN_TYPE_CASE_H
+#define SHOGUN_TYPE_CASE_H
+
+#include <typeindex>
+#include <unordered_map>
+
+#include <shogun/lib/SGMatrix.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/lib/any.h>
+#include <shogun/lib/type_list.h>
+
+using namespace shogun;
+
+namespace shogun
+{
+	typedef Types<
+		bool, char, int8_t, uint8_t, int16_t, uint16_t, int32_t, uint32_t,
+		int64_t, uint64_t, float32_t, float64_t, floatmax_t, SGVector<int32_t>,
+		SGVector<int64_t>, SGVector<float32_t>, SGVector<float64_t>,
+		SGVector<floatmax_t>, SGMatrix<int32_t>, SGMatrix<int64_t>,
```

Comment:
those should suffice for now, so nice :)

Reply:
Do we need std::vector too?

Reply:
Let's think. The feature of going from any to typed will be used in
* logging (only model parameters, numerical stuff)
* modelselecton (again, mostly model parameters, here we will need `SGObject` as well ...)
* not sure what else?
So with all that in mind, I think those types should be ok. And we can extend on request .... like done for any